src/jit/importer.cpp

```diff
@@ -16440,6 +16439,33 @@ bool Compiler::impReturnInstruction(BasicBlock* block, int prefixFlags, OPCODE&
 #endif
                 }
 
+                // If we are inlining a method that returns a struct byref, check whether we are "reinterpreting" the
+                // struct.
+                if ((returnType == TYP_BYREF) && (info.compRetType == TYP_BYREF) && (op2->OperGet() == GT_ADDR))
+                {
+                    GenTree* op2Child = op2->gtGetOp1();
+                    if ((op2Child->OperGet() == GT_LCL_VAR) && varTypeIsStruct(op2Child->TypeGet()))
```

Comment:
It appears that this doesn't impact the handling of SIMD types, presumably because any code that is looking at `lvOverlappingFields` is dealing with field accesses, which we don't expect to see for those. However, to avoid future pessimization, I think it would be better to change this to:
```
if ((op2Child->OperGet() == GT_LCL_VAR) && (op2Child->TypeGet() == TYP_STRUCT))
```

Reply:
Just to confirm: are we never expected to see field accesses for SIMD types (now or in the future)? If so, I'll change the check. If field accesses are potentially possible for SIMD types, we may run into the same bug for them.

Reply:
We sometimes access fields of the fixed-size SIMD types, so a more conservative fix might be better:
```
if ((op2Child->OperGet() == GT_LCL_VAR) && varTypeIsStruct(op2Child->TypeGet() &&
    !isOpaqueSIMDLclVar(lvaGetDesc(op2Child->AsLclVarCommon())))
```

Reply:
Done.